src/plugins/intel_cpu/src/transformations/snippets/x64/pass/lowered/brgemm_blocking.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "brgemm_blocking.hpp"
+
+#include "openvino/pass/pattern/matcher.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "snippets/itt.hpp"
+#include "snippets/lowered/linear_ir.hpp"
+#include "snippets/lowered/loop_manager.hpp"
+#include "snippets/snippets_isa.hpp"
+#include "transformations/snippets/x64/op/brgemm_cpu.hpp"
+
+
+namespace ov {
+namespace intel_cpu {
+namespace pass {
+using LoopManager = snippets::lowered::LinearIR::LoopManager;
+using LoopInfoPtr = LoopManager::LoopInfoPtr;
+using LoopPort = LoopManager::LoopPort;
+
+BrgemmBlocking::BrgemmBlocking() : Pass() {}
+
+bool BrgemmBlocking::run(snippets::lowered::LinearIR& linear_ir) {
+    OV_ITT_SCOPED_TASK(ov::pass::itt::domains::SnippetsTransform, "Snippets::BrgemmBlocking")
+    if (linear_ir.empty())
+        return false;
+
+    bool modified = false;
+    const auto& loop_manager = linear_ir.get_loop_manager();
+
+    auto brgemm_m = ov::pass::pattern::wrap_type<ov::intel_cpu::BrgemmCPU>();
+    auto matcher = std::make_shared<ov::pass::pattern::Matcher>(brgemm_m, "BrgemmLowered");
+    // TODO: make the block size configurable
+    const auto m_block_size = 32;
+
+    for (const auto& expr : linear_ir) {
+        const auto node = expr->get_node();
+        if (!matcher->match(node) || !expr->get_loop_ids().empty())
```

Comment:
In general we can have `Brgemm` in loop but this loop is not blocked (by batch dimension).
Could you please update this logic? Probable it should be so: `innermost_increment == brgemm->get_input_count(0)` where  `innermost_increment` is increment of the innermost Loop for `Brgemm`.
May I ask you to verify this moment? Thanks a lot!

Reply:
The logic has been updated: lambda `blocking_loop_exists` is responsible for this. The lambda compares increments and dimension index for all loops